src/audio/dai.c

```diff
@@ -64,6 +74,47 @@ struct dai_data {
 	uint64_t wallclock;	/* wall clock at stream start */
 };
 
+static void dai_atomic_trigger(void *arg, enum notify_id type, void *data);
+
+static struct dai_group *dai_add_group(struct comp_dev *dev, uint8_t group_id)
+{
+	struct list_item *dai_groups = &sof_get()->dai_groups;
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct list_item *group_item;
+	struct dai_group *group;
+
+	if (dd->group) {
+		comp_info(dev, "dai_add_group(), DAI already in group");
+		return dd->group;
+	}
+
+	list_for_item(group_item, dai_groups) {
+		group = container_of(group_item, struct dai_group, list);
+		if (group && group->group_id == group_id)
+			break;
+		else
+			group = NULL;
+	}
+
+	if (!group) {
+		group = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM,
+				sizeof(*group));
```

Comment:
test if allocation failed?

Reply:
If it fails it will result in immediate panic, since it's the system heap.

Reply:
ack